bridge/src/commonMain/kotlin/ru/herobrine1st/matrix/bridge/telegram/TelegramWorker.kt

```diff
@@ -82,6 +83,28 @@ class TelegramWorker(
                                 }
                             }
 
+                            is RoomMessageEventContent.FileBased.File -> {
+                                val caption = when {
+                                    content.fileName == null || content.fileName == content.body ->
+                                        "${event.sender} sent a file"
+
+                                    else -> "${event.sender}: ${content.body}"
+                                }
+
+                                val url = content.url ?: return
+                                val fileName = content.fileName ?: "file.bin"
```

Comment:
[`body` is the file name if `filename` is omitted](https://spec.matrix.org/v1.14/client-server-api/#mfile)
Max length of file name is 255 bytes (on linux). Allowing `/` and `\` characters leads to path traversal vulnerability on respective platforms.

https://github.com/kotlin-telegram-bot/kotlin-telegram-bot/pull/349 solves it all without involving filesystem. I think it's time to fork.

```suggestion
                                val fileName = (content.fileName ?: content.body).take(255)
                                    .replace('/', '_')
                                    .replace('\\', '_')
```